torch/csrc/autograd/engine.cpp

```diff
@@ -495,9 +498,10 @@ static variable_list call_function(NodeTask& task) {
     // If you change the logic here, make sure it's compatible with accumulate_grad.cpp.
     auto inputs_copy = inputs;
     outputs = fn(std::move(inputs_copy));
-  }else{
+  } else {
     outputs = fn(std::move(inputs));
   }
+  at::setThreadLocalDebugInfo(prev_info);
```

Comment:
This doesn't look exception safe.

Reply:
I'll mark the API as noexcept

Reply:
As in use RAII

Reply:
sounds good, let me add a guard object to use